tests/e2e/cucumber/steps/ui/spaces.ts

```diff
@@ -117,22 +117,34 @@ When(
 )
 
 Then(
-  '{string} should see folder {string} but should not be able to edit',
-  async function (this: World, stepUser: string, resource: string): Promise<void> {
+  /^"([^"]*)" (should|should not) be able to edit (?:folder|file) "([^"]*)"$/,
+  async function (
+    this: World,
+    stepUser: string,
+    actionType: string,
+    resource: string
+  ): Promise<void> {
     const { page } = this.actorsEnvironment.getActor({ key: stepUser })
     const spacesObject = new objects.applicationFiles.Spaces({ page })
     const userCanEdit = await spacesObject.canUserEditResource({ resource })
-    expect(userCanEdit).toBe(false)
+    expect(userCanEdit).toBe(actionType === 'should' ? true : false)
   }
 )
 
 Then(
-  '{string} should see file {string} but should not be able to edit',
-  async function (this: World, stepUser: string, resource: string): Promise<void> {
+  /^"([^"]*)" (should|should not) see (show links|show invited people) button on the (?:folder|file) "([^"]*)"$/,
+  async function (
+    this: World,
+    stepUser: string,
+    actionType: string,
+    shareType: string,
+    resource: string
+  ): Promise<void> {
     const { page } = this.actorsEnvironment.getActor({ key: stepUser })
     const spacesObject = new objects.applicationFiles.Spaces({ page })
-    const userCanEdit = await spacesObject.canUserEditResource({ resource })
-    expect(userCanEdit).toBe(false)
+    actionType === 'should'
+      ? await spacesObject.expectThatShowShareButtonExist({ shareType, resource })
+      : await spacesObject.expectThatShowShareButtonNotExist({ shareType, resource })
```

Comment:
Can we do like this: get the locator and assert here in step def
```suggestion
    const showShareButtonSelector = await spacesObject.getShowShareButtonSelector({ buttonLabel, resource })
    actionType === 'should'
      ? await expect(locator).toBeVisible()
      : await expect(locator).not.toBeVisible()
```

Reply:
I tried to avoid `page.locator('button').isVisible()` and use 
`await expect(page.locator('button')).toBeVisible()` instead.

but your suggestion is fine 👍️